### DISCUSSION TITLE
Get the public URL of an image?

### DISCUSSION THREAD

**Discussion by fostimus** [gatsbyjs/gatsby#36361] [Help] [answered]
I'm attempting to set `<meta name="twitter:image" content="<MY IMG URL HERE>" />` for my site. The image URL needs to be absolute, so I can't just set the path to the image and have it work.

I've tried many things, but can't seem to get a public URL to my image. I tried using the static folder, which the docs says will directly put the image in `/public` at run time, but that isn't the case. I'm able to get a publicURL to images for blog posts, when we process them through remark.

How can I get the public URL for any image outside of remark?

**Comment by tyhopp** [comment, marked as answer]
You could try prepending `siteUrl` from your site metadata before the path where your image lives if you need it to be absolute. Check out the snippet in https://www.gatsbyjs.com/docs/how-to/adding-common-features/adding-seo-component/#seo-component for an example.

You could also add conditional logic in your `gatsby-config` to set `siteUrl` based on your environment if you need that too.

**Reply by fostimus** [reply]
Thanks for the reply! Is there a way to do this with images that aren't set under `siteMetadata`?

For context, this is for SEO purposes, but it's for setting the twitter image of landing pages.

**Reply by TylerBarnes** [reply]
@fostimus the images don't need to be set there, just the frontend url. I shortened the example above for you and added a comment:

```js
import React from "react"
import { useSiteMetadata } from "../hooks/use-site-metadata"

export const SEO = ({ children }) => {
  const { siteUrl, image } = useSiteMetadata()
  const seo = {
    image: `${siteUrl}${image}`, // <-- this image path could come from anywhere, you can make other queries with useStaticQuery()
  }
  return (
    <>
      <meta name="image" content={seo.image} />
      {children}
    </>
  )
}
```
